src/test/java/dal/fatih/todoly/TodolyTest.java

```diff
@@ -45,66 +51,71 @@
     private TaskRepository taskRepository;
 
     @Test
-    public void shouldCreateTask() throws URISyntaxException {
+    public void shouldCreateTask() throws URISyntaxException, JsonProcessingException {
         URI taskCreateUrl = new URI(createURLWithPort("/task"));
-        HttpEntity<TaskDTO> request = new HttpEntity<TaskDTO>(
-                new TaskDTO("Created-title-of-task", "Created-description-of-task"
-                        , LocalDateTime.now().plusDays(10))
-        );
+        LocalDateTime dueDate = LocalDateTime.parse(LocalDateTime.now().plusDays(10).format(dateTimeFormat));
+        TaskDTO taskDto = new TaskDTO("Created-title-of-task", "Created-description-of-task"
+                , dueDate);
+
+        HttpEntity<TaskDTO> request = new HttpEntity<>(taskDto);
 
         ResponseEntity<String> responseEntity = this.testRestTemplate
                 .postForEntity(taskCreateUrl, request, String.class);
-        logger.warn(responseEntity.getBody());
 
         String response = responseEntity.getBody();
-        assert response != null;
-        long createdTaskId = Long.parseLong(response.substring(6, response.length() - 1));
 
-        String actual = taskRepository.get(createdTaskId).getTitle();
-        String expected = "Created-title-of-task";
+        Map<String, Long> map = jsonMapper.readValue(response, new TypeReference<Map<String, Long>>() {
+        });
+        long createdTaskId = map.get("id");
 
-        assertThat(actual, is(equalTo(expected)));
+        Task actual = taskRepository.get(createdTaskId);
+
+        assertThat(actual, is(notNullValue()));
+        assertThat(actual.getTitle(), equalTo(taskDto.getTitle()));
+        assertThat(actual.getDescription(), equalTo(taskDto.getDescription()));
+        assertThat(actual.getDueDate(), equalTo(taskDto.getDueDate()));
         assertThat(responseEntity.getStatusCode(), equalTo(HttpStatus.CREATED));

```

Comment:
Move this assertion right after calling `postForEntity` method. It is good to fail the test earlier if the status is not even healthy so that you don't deserialize the payload or anything.